packages/atomic-hosted-page/src/autoloader/index.ts

```diff
@@ -62,9 +62,22 @@ if (typeof window !== 'undefined') {
       }
     }
   });
-  // Initial discovery
-  discover(document.body);
-  // Listen for new undefined elements
-  observer.observe(document.documentElement, {subtree: true, childList: true});
-  //# sourceMappingURL=index.js.map
+
+  const initializeDiscovery = () => {
+    // Initial discovery
+    discover(document.body);
+    // Listen for new undefined elements
+    observer.observe(document.documentElement, {
+      subtree: true,
+      childList: true,
+    });
+  };
+
+  // https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event#checking_whether_loading_is_already_complete
+  if (document.readyState === 'loading') {
```

Comment:
Is this check really necessary?

My understanding is that the script that loads `atomic-hosted-page.esm.js` from the CDN would already deferred by default because it's a module, which means that the script will necessarily execute after the document has been parsed but before `DOMContentLoaded` is fired.

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/script#defer

Reply:
I think it's best not to make too many assumptions about customers loading the library from the CDN.
For example, they may load it using asynchronous import (`import()`) or put the async attributes on the script.

What caused us to detect the issue was just that: making too many assumptions about how the lib was loaded.